parallel-install/example/example.go

```diff
@@ -71,14 +75,32 @@ func main() {
 
 	config.SetupLogger(log.Printf)
 
-	err = installer.StartKymaInstallation(restConfig)
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		log.Printf("Failed to create kube client. Exiting...")
+		os.Exit(1)
+	}
+
+	overridesProvider, err := overrides.New(kubeClient, installer.OverridesYamls, installer.Cfg.Log)
```

Comment:
All this logic should be internal to the library. The consumer of the library should not care for overrides provider, prerequisite provider, engine creation, etc. The consumer should be able to create an installation object with the configuration and then call the installation and uninstallation functions.

Reply:
I'm afraid that if we move this back we will not be able to test it at all. I understand your concerns and I don't like this solution, too, but after we have tests we can refactor this code to make it more user-friendly and still be sure that everything works as expected.

Reply:
@suleymanakbas91 I can agree with the exception of kubeClient - this should be provided by the user, I don't see how it should be library responsibility to create a k8s client. It should be an external dependency.